src/cpp/src/clustering.cpp

```diff
@@ -27,6 +33,51 @@ shared_ptr<Clustering> kmeans(Tensor vectors,
     int n = vectors.size(0);
     int d = vectors.size(1);
 
+    #ifdef QUAKE_ENABLE_GPU
+    bool use_gpu = true;
```

Comment:
Put use_gpu as an argument to kmeans (default false). It's best to not have too many #ifdefs in the code